azkaban-common/src/main/java/azkaban/project/DirectoryYamlFlowLoader.java

```diff
@@ -40,6 +44,7 @@
   private final Props props;
   private final Set<String> errors = new HashSet<>();
```

Comment:
Question: I'm seeing there exist a few common variables in class DirectoryYamlFlowLoader and class DirectoryFlowLoader, why not make FlowLoader an Abstract class rather than Interface?

Reply:
Good question. Several reasons why interface is a better option here:
1. The two classes don't share common implementations. They need to implement their own method `loadProjectFLow` in their own different ways. So defining the method in the interface should be sufficient.
2. Some variables might be common to the two classes but some are not. If we need to add a new type of FlowLoader, it won't necessarily share the same variables. In this case, making it abstract class for sharing common variables might not be flexible.
Actually in my previous PR, I used Abstract class first and then switched to interface. You can refer to the discussions below:
https://github.com/azkaban/azkaban/pull/1500/commits/30744ec9e375f03ce12525e64b50d2c8b1899f21#diff-33dbf7e0198cf8b60e328d5fde7921acR35